### PR TITLE
refactor: Update subscriber to use Kubernetes patch for ConfigMaps an…

### PR DESCRIPTION
**Pull request by RipulHandoo**
…d Secrets instead of get-update

## Proposed changes
In the ClusterRegister function within [operations.go](https://github.com/litmuschaos/litmus/blob/master/litmus-portal/cluster-agents/subscriber/pkg/k8s/operations.go), I've replaced the 'update' method with 'kubernetes patch' for updating ConfigMaps and Secrets. This change involves using 'json.Marshal' to serialize the clusterData, which is then utilized in 'Patch' with 'MergePatchType' for both ConfigMap and Secret updates.

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

## Special notes for your reviewer:
This PR solve's this #3944

### CODE REVIEW

**Review by Saranya-jena — CHANGES_REQUESTED** [review]
Hey @RipulHandoo you would need to add the changes in `chaoscenter` directory since with the release of Litmus 3.0, litmus-portal dir has been deprecated.

**Comment by RipulHandoo** [comment]
> Hey @RipulHandoo you would need to add the changes in `chaoscenter` directory since with the release of Litmus 3.0, litmus-portal dir has been deprecated.

Sure, I'll make the necessary changes in [this file](https://github.com/litmuschaos/litmus/blob/master/chaoscenter/subscriber/pkg/k8s/operations.go#L158) shortly.

**Comment by RipulHandoo** [comment]
> Hey @RipulHandoo you would need to add the changes in `chaoscenter` directory since with the release of Litmus 3.0, litmus-portal dir has been deprecated.

Hey @Saranya-jena, I have made necessary changes in the `chaoscenter` . Please review and let me know if everything is okay.

File: chaoscenter/subscriber/pkg/k8s/operations.go [hunk 1]
```diff
@@ -155,51 +157,63 @@ func (k8s *k8sSubscriber) IsAgentConfirmed() (bool, string, error) {
...
-		"INFRA_SCOPE":        infraData["INFRA_SCOPE"],
-		"COMPONENTS":         infraData["COMPONENTS"],
-		"START_TIME":         infraData["START_TIME"],
-		"VERSION":            infraData["VERSION"],
-		"SKIP_SSL_VERIFY":    infraData["SKIP_SSL_VERIFY"],
-		"CUSTOM_TLS_CERT":    infraData["CUSTOM_TLS_CERT"],
+	// Define a patch object for the ConfigMap with only "IS_CLUSTER_CONFIRMED".
+	newConfigMapData := map[string]interface{}{
+		"data": map[string]interface{}{
+			"IS_CLUSTER_CONFIRMED": true,
```
Comment (Saranya-jena) [inline chaoscenter/subscriber/pkg/k8s/operations.go:169]: This should be `IS_INFRA_CONFIRMED`

File: chaoscenter/subscriber/pkg/k8s/operations.go [hunk 2]
```diff
@@ -11,7 +12,7 @@ import (
 
 	"k8s.io/apimachinery/pkg/labels"
 
-	"subscriber/pkg/types"
+	atype "subscriber/pkg/types"
```
Comment (Saranya-jena) [inline chaoscenter/subscriber/pkg/k8s/operations.go:15]: please give it a meaningful name

**Review by Saranya-jena — CHANGES_REQUESTED** [review]

**Review by amityt — APPROVED** [review]

File: chaoscenter/subscriber/pkg/k8s/operations.go [hunk 3]
```diff
@@ -155,51 +157,63 @@ func (k8s *k8sSubscriber) IsAgentConfirmed() (bool, string, error) {
...
+		types.StrategicMergePatchType,
+		configMapPatchBytes,
+		metav1.PatchOptions{},
+	)
 	if err != nil {
 		return false, err
 	}
 
-	logrus.Info(InfraConfigName + " has been updated")
+	logrus.Info("%s has been updtaed", InfraConfigName)
```
Comment (Jonsy13) [inline chaoscenter/subscriber/pkg/k8s/operations.go:191]:
```suggestion
	logrus.Info("%s has been updated", InfraConfigName)
```

File: chaoscenter/subscriber/pkg/k8s/operations.go (hunk 1 above)
Comment (RipulHandoo) [reply chaoscenter/subscriber/pkg/k8s/operations.go]: Corrected.

**Review by Jonsy13 — APPROVED** [review]

**Review by Saranya-jena — APPROVED** [review]

**Review by SarthakJain26 — APPROVED** [review]

**Pull request merged**